extensions/console-notification.crd.yaml

```diff
@@ -0,0 +1,61 @@
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: consolenotifications.config.openshift.io
+  annotations:
+    displayName: ConsoleNotification
+    description: Extension for configuring openshift web console notifications.
+spec:
+  group: config.openshift.io
+  version: v1
+  versions:
+  - name: v1
+    served: true
+    storage: true
+  scope: Cluster
+  names:
+    plural: consolenotifications
+    singular: consolenotification
+    kind: ConsoleNotification
+    listKind: ConsoleNotificationList
+  additionalPrinterColumns:
+  - name: Age
+    type: date
+    JSONPath: .metadata.creationTimestamp
+  subresources:
+    status: {}
+  validation:
+    openAPIV3Schema:
+      properties:
+        spec:
+          type: object
+          description: Represents console notification spec
+          required:
+          - notification
+          - scope
+          properties:
+            notification:
+              type: object
+              description: Object that holds notification extension
+              required:
+                - text
+              properties:
+                text:
+                  type: string
+                  description: Text of the notification
+                link:
+                  type: object
+                  description: Object that holds notification link details
+                  required:
+                    - href
+                  properties:
+                    href:
+                      type: string
+                      description: HREF for the link
+                    text:
```

Comment:
Possibly a `target` option as well? If the notification is a link out to another tool, do we want the option for a new tab so as not to lose console context?

Reply:
I personally don't think the use of `target="_blank"` is good (see https://www.nngroup.com/articles/the-top-ten-web-design-mistakes-of-1999/), but I will add the feature if others think it worthwhile.

Reply:
@alimobrem, @robszumski, @openshift/team-ux-review, please chime in here.  

To put a finer point on why I disagree with opening links in a new window:
1.  it breaks the back button (see [#1](https://www.nngroup.com/articles/the-top-ten-web-design-mistakes-of-1999/))
1.  it's a user-hostile action.  IMO, the user should decide where links open.  (see [#2](https://www.nngroup.com/articles/the-top-ten-web-design-mistakes-of-1999/))

That said, we open links in a new window elsewhere in the console, and providing the option to open links in a new window seems reasonable.  That said, once we've added the option, it's much harder to take it away.

Reply:
Currently, we open all external links in a new window. I'd think we'd want these links to always open in a new window (and skip the option). I realize there's debate on this, but it would be consistent with current behavior :)

Reply:
Worth noting, we are not currently requiring these links be absolute.  Both `/` and `http://www.example.com` are valid, so always opening in a new window isn't coupled to whether or not the link is external.

Reply:
We probably need some validation. At the very least, we can't accept relative URLs since the URL for every page is different (and the banner is visible on every page).

Reply:
I lost the link to the BZ when I closed bluejueans. Is having a link in that message even a customer requirement at this point?